applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -275,42 +292,50 @@ static bool app_event_handler(const struct app_event_header *aeh)
 		return false;
 	}
 
-	if (IS_ENABLED(CONFIG_CAF_BLE_COMMON_EVENTS) && is_ble_peer_conn_params_event(aeh)) {
+	if (IS_ENABLED(CONFIG_DESKTOP_DVFS_STATE_LLPM_CONNECTED_ENABLE) &&
+	    is_ble_peer_conn_params_event(aeh)) {
 		return handle_ble_peer_conn_params_event(cast_ble_peer_conn_params_event(aeh));
 	}
 
-	if (IS_ENABLED(CONFIG_CAF_BLE_COMMON_EVENTS) && is_ble_peer_event(aeh)) {
+	if (IS_ENABLED(CONFIG_DESKTOP_DVFS_STATE_LLPM_CONNECTED_ENABLE) &&
+	    is_ble_peer_event(aeh)) {
 		if (cast_ble_peer_event(aeh)->state == PEER_STATE_DISCONNECTED) {
 			process_dvfs_states(DVFS_STATE_LLPM_CONNECTED, false);
 		}
 
 		return false;
 	}
 
-	if (IS_ENABLED(CONFIG_DESKTOP_USB_ENABLE) && is_usb_state_event(aeh)) {
+	if (IS_ENABLED(CONFIG_DESKTOP_DVFS_STATE_USB_CONNECTED_ENABLE) &&
+	    is_usb_state_event(aeh)) {
 		const struct usb_state_event *event = cast_usb_state_event(aeh);
 
 		process_dvfs_states(DVFS_STATE_USB_CONNECTED, event->state == USB_STATE_ACTIVE);
 
 		return false;
 	}
 
-	if (IS_ENABLED(CONFIG_CAF_BLE_SMP_TRANSFER_EVENTS) && is_ble_smp_transfer_event(aeh)) {
+	if (IS_ENABLED(CONFIG_DESKTOP_DVFS_STATE_SMP_TRANSFER_ENABLE) &&
+	    is_ble_smp_transfer_event(aeh)) {
 		process_dvfs_states(DVFS_STATE_SMP_TRANSFER, true);
 		struct dvfs_state_timeout *smp_transfer_timeout =
 			&dvfs_state_timeouts[DVFS_STATE_SMP_TRANSFER];
-		(void) k_work_reschedule(&(smp_transfer_timeout->timeout_work),
-					 K_MSEC(smp_transfer_timeout->timeout_ms));
+		if (smp_transfer_timeout->timeout_ms > 0) {
```

Comment:
If the user sets the timeout to zero, then this DVFS state will never be deactivated